vortex-io/src/object_store.rs

```diff
@@ -124,40 +88,39 @@ impl VortexReadAt for ObjectStoreReadAt {
 }
 
 pub struct ObjectStoreWriter {
-    multipart: Option<WriteMultipart>,
+    upload: Box<dyn MultipartUpload>,
 }
 
 impl ObjectStoreWriter {
-    pub fn new(multipart: WriteMultipart) -> Self {
-        Self {
-            multipart: Some(multipart),
-        }
+    pub async fn new(object_store: Arc<dyn ObjectStore>, location: Path) -> VortexResult<Self> {
+        let upload = object_store.put_multipart(&location).await?;
+        Ok(Self { upload })
     }
 }
 
 impl VortexWrite for ObjectStoreWriter {
     async fn write_all<B: IoBuf>(&mut self, buffer: B) -> io::Result<B> {
-        self.multipart
-            .as_mut()
-            .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "multipart already finished"))
-            .map(|mp| mp.write(buffer.as_slice()))?;
+        let slice = buffer.as_slice();
+
+        const CHUNKS_SIZE: usize = 25 * 1024 * 1204;
+
+        for (chunk_idx, chunk) in slice.chunks(25 * 1024 * 1204).enumerate() {
+            let payload = Bytes::copy_from_slice(&slice[chunk_idx * CHUNKS_SIZE..][..chunk.len()]);
```

Comment:
why not use `chunk` here instead of re-slicing the slice?

Reply:
I thought I had a lifetime issue, apparently I don't